app/scenes/Search/Search.js

```diff
@@ -33,349 +28,352 @@ import DateFilter from "./components/DateFilter";
-      pathname: this.props.location.pathname,
-      search: queryString.stringify({
-        ...queryString.parse(this.props.location.search),
-        ...search,
-      }),
-    });
-  };
-
-  handleNewDoc = () => {
-    if (this.collectionId) {
-      this.props.history.push(newDocumentUrl(this.collectionId));
-    }
-  };
-
-  get includeArchived() {
-    return this.params.get("includeArchived") === "true";
-  }
-
-  get collectionId() {
-    const id = this.params.get("collectionId");
+function Search({ notFound }: Props) {
+  const { documents } = useStores();
+  const location = useLocation<LocationWithState>();
+  const history = useHistory();
+  const match = useRouteMatch();
+
+  const { t } = useTranslation();
+
+  let firstDocument = React.useRef<React.Component<any>>(null);
+  const lastQuery = React.useRef("");
+  const lastParams = React.useRef<Object>({});
+
+  const [query, setQuery] = React.useState<string>(
+    decodeURIComponentSafe(match.params.term || "")
+  );
+  const [params, setParams] = React.useState<URLSearchParams>(
+    new URLSearchParams()
+  );
+  const [offset, setOffset] = React.useState(0);
+  const [allowLoadMore, setAllowLoadMore] = React.useState(true);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [pinToTop, setPinToTop] = React.useState(!!match.params.term);
+
+  const locationSearchRef = React.useRef(location.search);
+  const matchParamsTermRef = React.useRef(match.params.term);
+
+  const [results, setResults] = React.useState([]);
+  const [showEmpty] = React.useState(
+    !isLoading && query && results.length === 0
+  );
+  const [showShortcutTip] = React.useState(
+    !pinToTop && location.state && location.state.fromMenu
+  );
+
+  const collectionId = React.useCallback(() => {
+    const id = params.get("collectionId");
     return id ? id : undefined;
-  }
+  }, [params]);
 
-  get userId() {
-    const id = this.params.get("userId");
+  const dateFilter = React.useCallback(() => {
+    const id = params.get("dateFilter");
     return id ? id : undefined;
-  }
+  }, [params]);
 
-  get dateFilter() {
-    const id = this.params.get("dateFilter");
+  const userId = React.useCallback(() => {
+    const id = params.get("userId");
     return id ? id : undefined;
-  }
-
-  get isFiltered() {
-    return (
-      this.dateFilter ||
-      this.userId ||
-      this.collectionId ||
-      this.includeArchived
-    );
-  }
-
-  get title() {
-    const query = this.query;
-    const title = this.props.t("Search");
-    if (query) return `${query} – ${title}`;
-    return title;
-  }
-
-  @action
-  loadMoreResults = async () => {
-    // Don't paginate if there aren't more results or we’re in the middle of fetching
-    if (!this.allowLoadMore || this.isLoading) return;
+  }, [params]);
 
-    // Fetch more results
-    await this.fetchResults();
-  };
+  const includeArchived = React.useCallback(() => {
+    return params.get("includeArchived") === "true";
+  }, [params]);
```

Comment:
These could all be simple `const` now, eg:

```
const includeArchived = params.get("includeArchived") === "true";
```

Same applies for the other params